### DISCUSSION TITLE
Salt Edge API Key

### DISCUSSION THREAD

**Discussion by taker218** [firefly-iii/firefly-iii#5451] [Firefly III] [answered]
Hello everyone,

I just came across this tool and really like it so far.
I wanted to use the Salt Edge importer and requested a test access.
In the mail they wrote the following:

"In case the project is solely for personal use, I would like to suggest you to use our partners platform: FireFly III
 https://www.firefly-iii.org/
 On their website you will find a detailed guide on how to get unlimited free access for personal use with Salt Edge API."
 
 I have looked at the documentation, but I couldn't find anything regarding that topic.

Could anyone point me in the right direction?

Best regards,
Thomas

**Comment by JC5** [comment, marked as answer]
Mmm, there's no other documentation. Or at least, not much. You need to get test access from them and indicate that you use Firefly III. That should be enough, so you can reply "I am using Firefly III".

To import data, you need the Data Importer: https://docs.firefly-iii.org/data-importer/

**Comment by taker218** [comment]
Thanks, I already have the importer running, just need to get the access.
I will write them that I'm going to use firefly :)

**Comment by MacPaille** [comment]
I got the test account on SaltEdge without even writing to them, just filling the required info on the website. But now I have it I still can't add a connection to any real bank. Not sure what is wrong

**Reply by JC5** [reply]
Make sure you delete all old connections, then try to create a new one. Could make the difference.

**Reply by MacPaille** [reply]
I have no connections anymore as it was all to their fake country. I got a reply from the support ticket and I have to add the provider keys. But even after having done that it does not work... And I saw a lot of info with having to register for the banks APIs, provide valid certificates etc...

Compared to Nordigen which use the customer authentication like I'm using to access my bank portal this is much easier... Dont know why that does not work on Salt Edge...

**Comment by taker218** [comment]
I just got the access for private use and it seems like there is no time limit on it, will see if it still works in 3 months :)

**Reply by MacPaille** [reply]
The test access? Indeed the only limit is 10 connections. But I'm not able to create the connections to my banks as this needs to add the provider keys along with eIDAS certificates...
Are you able to make it work?

I find Nordigen much easier on this

**Reply by taker218** [reply]
Yes, I just did it via the importer and added a new connection.
Just needed to input my online banking credentials and got all the transactions from the last 90 days (which is the standard of my bank).

**Reply by MacPaille** [reply]
Ho come it does not work for me :(
In your Settings > Application, what have you selected for "What products do you plan to use within Salt Edge API?" ?

Or have you used Dynamic registrations?

**Reply by taker218** [reply]
I have "Account Balance Check".

I had to add a service key API to use it, with the App key the connection didn't work for me.

**Reply by MacPaille** [reply]
Strange... Did you have to generate a certificate and sign it somewhere? Seems my error is saying invalid private key...

**Reply by taker218** [reply]
no, I just added the new connection, chose my bank and input my credentials.
After that I just had to select the tan generator and after that the transactions were updated.

**Reply by MacPaille** [reply]
I suppose my test account is not set correctly then... How did you proceed to get it activated? Just sent them an email?

**Reply by taker218** [reply]
I just requested the test acces and got a mail from them after that.
But I suppose you could insert in the description, that you are using Firefly III so you would get the access.

**Reply by MacPaille** [reply]
I'm trying to create another account. Also have you added the connection from data-importer or from SaltEdge portal?

**Reply by taker218** [reply]
from the data importer directly

**Reply by MacPaille** [reply]
Now my new account is in test I4m still getting the same errors after having manually added my banks in the provider lists. Still complaining about invalid private key.
Have you setup anything else than just added your banks in the providers? Like Mutual SSL cert or something?

**Reply by taker218** [reply]
no, I just added a new connection via the importer.
I didn't have to do anything on the salt edge website other then getting the api key.

You might be better off opening a new discussion here, maybe someone can help you that way.
